switchbot/__init__.py

```diff
@@ -111,12 +115,41 @@ def _sendcommand(self, key, retry) -> bool:
         if send_success:
             return True
         if retry < 1:
-            _LOGGER.error("Switchbot communication failed. Stopping trying.", exc_info=True)
+            _LOGGER.error(
+                "Switchbot communication failed. Stopping trying.", exc_info=True
+            )
             return False
-        _LOGGER.warning("Cannot connect to Switchbot. Retrying (remaining: %d)...", retry)
+        _LOGGER.warning(
+            "Cannot connect to Switchbot. Retrying (remaining: %d)...", retry
+        )
         time.sleep(DEFAULT_RETRY_TIMEOUT)
         return self._sendcommand(key, retry - 1)
 
+    def get_servicedata(self, retry=DEFAULT_RETRY_COUNT, scan_timeout=5) -> None:
+        """Get BTLE 16b Service Data,
+        returns after the given timeout period in seconds."""
+        waiting_time = self._time_between_update_command - time.time()
+        if waiting_time > 0:
+            time.sleep(waiting_time)
+        try:
+            devices = bluepy.btle.Scanner().scan(scan_timeout)
+
+        except bluepy.btle.BTLEManagementError:
+            _LOGGER.warning("Error updateing Switchbot.", exc_info=True)
+
+        if not devices:
+            time.sleep(DEFAULT_RETRY_TIMEOUT)
+            return self.get_servicedata(retry - 1, scan_timeout)
```

Comment:
This will retry forever?
Check if retry > 0

Reply:
Done. Also added logging.